nixos/modules/services/web-apps/matomo.nix

```diff
@@ -183,7 +183,10 @@ in {
             chmod -R u+rwX,g+rwX,o-rwx "${dataDir}"
 
             # check whether user setup has already been done
-            if test -f "${dataDir}/config/config.ini.php"; then
+            if test -f "${dataDir}/config/config.ini.php" &&
+              # since matomo-5.2.0, the config.ini.php is already created at first
```

Comment:
On second thought: it seems matomo 4 is still the default on NixOS stable. Would this extra condition prevent pending DB upgrades from continuing in that version?

Reply:
No, that is unproblematic for version 4. Matomo 4 only creates the config.ini.php file after a successful installation, and that file then already contains the database section. So this execution guard works correctly.